sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointPosition.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Consumer
+{
+    /// <summary>
+    ///   The position in a partition's event stream to use when updating a checkpoint, indicating that an event processor should begin reading from the next event.
+    /// </summary>
+    ///
+    public struct CheckpointPosition
+    {
+        /// <summary>
+        ///   The offset to associate with the checkpoint. This value will be used for informational metadata in almost all cases. If there is no sequence number associated
+        ///   with the checkpoint, this value will be used for positioning when events are read.
+        /// </summary>
+        ///
+        public long? Offset { get; internal set; }
```

Comment:
We generally want to ensure that structs are immutable.   Can we remove the setters?

Reply:
I was working on updating this, but before we decided to use long.MinValue as the default sequence number value, I was using this a workaround for the old UpdateCheckpoint overload to call the new one, since we don't want to have a CheckpointPosition(long? sequenceNumber, long? offset) constructor